src/notification_service.py

```diff
@@ -0,0 +1,46 @@
+import dbus
+from dbus.mainloop.glib import DBusGMainLoop
+
+class Singleton(type):
+    __instances__ = {}
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls.__instances__:
+            cls.__instances__[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls.__instances__[cls]
+
+class NotificationService(metaclass=Singleton):
+
+    def __init__(self):
+        self.btsvc = None
+        self.monitor_bus = None
+        self.device = None
+
+    def start_service(self, device, service):
+
+        DBusGMainLoop(set_as_default=True)
+        self.monitor_bus = dbus.SessionBus()
+        self.monitor_bus.add_match_string_non_blocking("eavesdrop=true, interface='org.freedesktop.Notifications', member='Notify'")
```

Comment:
Match rules with "eavsdrop=true" are deprecated and dbus will drop them silently bases on the bus policy. You also need the right Flatpak sandbox permissions, namely "--socket=session-bus" or the appropriate "--talk-name=<dbus-name>" entries in the manifests finish-args segment. (Does your implementation actually work without them on your machine?)

Here is the modern implementation using the BecomeMonitor method:
```suggestion
        self.monitor_bus = dbus.SessionBus(private=True)
        try:
            self.dbus_monitor_iface = dbus.Interface(self.monitor_bus.get_object('org.freedesktop.DBus', '/org/freedesktop/DBus'),
                                                      dbus_interface='org.freedesktop.DBus.Monitoring')
            self.dbus_monitor_iface.BecomeMonitor(["interface='org.freedesktop.Notifications', member='Notify'"], 0)
        except dbus.exceptions.DBusException as e:
            print(e)
```

Reply:
I didn't notice that, rather I took the current daemon implementation. That makes a lot more sense though. 

It does work on my machine, but looking back, that is most likely because I changed a bunch of permissions in flatseal when working out the daemon version. I'll test it out with your version and add/reset those permissions.